algorithms/CPlusPlus/Arrays/fractional-knapsack.cpp

```diff
@@ -0,0 +1,77 @@
+// Description : Given weights and values of n items, put these items in a knapsack of capacity W to get the maximum total value in the knapsack.
+// We can break items for maximizing the total value of knapsack/
+
+// Algorithm Type: Divide & Conquer
+// Time Complexity: O(n*log(n))
+#include <bits/stdc++.h>
+using namespace std;
+class Item
+{
+public:
+    int value, weight;
+
+    // Constructor
+    Item(int value, int weight)
+    {
+        this->value = value;
+        this->weight = weight;
+    }
+};
+bool comp(Item a, Item b)
+{
+    double r1 = (double)a.value / (double)a.weight;
+    double r2 = (double)b.value / (double)b.weight;
+    return r1 > r2;
+}
+
+//Function to get the maximum total value in the knapsack.
+double fractionalKnapsack(int W, Item arr[], int n)
+{
+
+    sort(arr, arr + n, comp);
+
+    int curWeight = 0;
+    double finalvalue = 0.0;
+
+    for (int i = 0; i < n; i++)
+    {
+
+        if (curWeight + arr[i].weight <= W)
+        {
+            curWeight += arr[i].weight;
+            finalvalue += arr[i].value;
+        }
+
+        else
+        {
+            int remain = W - curWeight;
+            finalvalue += (arr[i].value / (double)arr[i].weight) * (double)remain;
+            break;
+        }
+    }
+
+    return finalvalue;
+}
+int main()
+{
+    int W; //    Weight of knapsack
+    cout << "Enter the weight(W) of the knapsack" << endl;
+    cin >> W;
+    int n;
+    cout << "Enter the total number(n) of {Value-Weight}pairs" << endl;
+    cin >> n;
+    Item arr[n];

```

Comment:
VLA ( Variable Length Arrays) are not supported by C++ standards, you are lucky that your compiler(most probably gcc) is supporting it as an extension.

Reply:
this not only makes your code non portable, but also make is dangerous as it opens a potential risk of having a huge array on your memory stack, making your code more venerable to stack overflow. 

It's much better to either use stl alternative ( like std::vector ) or if you are feeling brave then manually allocate ( and manage ) dynamically allocated array.